### ISSUE TITLE
hard dependency on devtools

### ISSUE THREAD

**Issue by Brunwo** [tschuehly/spring-view-component#26]
the dependency graph for  spring-view-component-core shows devtools : 

|    +--- de.tschuehly:spring-view-component-core:0.8.1
|    |    +--- org.springframework.boot:spring-boot-starter-web -> 3.2.5 (*)
|    |    +--- org.springframework.boot:spring-boot-starter-aop -> 3.2.5 (*)
|    |    +--- org.springframework.boot:spring-boot-devtools -> 3.2.5
|    |    |    +--- org.springframework.boot:spring-boot:3.2.5 (*)
|    |    |    \--- org.springframework.boot:spring-boot-autoconfigure:3.2.5 (*)

https://github.com/tschuehly/spring-view-component/blob/3ea2a9af8d4c68423526d7d025a27b830dc2c0f1/core/build.gradle.kts#L28

I"m pointing this because I wanted to run your lab repository, that relies on this, but had runtime clashing classloader issues, couldn't remove devtools to try to fix

**Comment by tschuehly** [comment]
Yeah the livereload implementation relies on devtools, I don't know how to make it optional tho. 
Have to figure that out sometime

**Comment by Brunwo** [comment]
could replace 'implementation' by developmentOnly , unless you want it also present for launches other than bootrun

**Comment by tschuehly** [comment]
> could replace 'implementation' by developmentOnly , unless you want it also present for launches other than bootrun

The problem is that the https://github.com/tschuehly/spring-view-component/blob/master/core/src/main/kotlin/de/tschuehly/spring/viewcomponent/core/processor/ViewComponentChangeListener.kt directly depends on devtools

**Comment by Clickin** [comment]
This problem can be solved with checked exception. Maybe it can throw more clear exception message.
